### PR TITLE
Rvr 2369 Refactor events handling (#9)

### PR DESCRIPTION
**Pull request by AlessandroDG**
## Type of change
- [x] Feature
- [x] Refactoring (no functional changes, no api changes)

## Description of change

Refactor rivrAnalyticsAdapter.js events handling.
Add ADD_AD_UNITS to trackable events.

### CODE REVIEW

**Comment by AlessandroDG** [comment]
On circle-ci tests pass but the job times out.
I don't seem to have permissions to rebuild it, let me know if you have any suggestions.

From circle-ci:
```Chrome 62.0.3202 (Windows 10.0.0): Executed 3469 of 3470 (skipped 1) SUCCESS (8.369 secs / 3.31 secs)
IE 11.0.0 (Windows 10.0.0): Executed 3466 of 3467 (skipped 1) SUCCESS (8.551 secs / 3.916 secs)
Firefox 58.0.0 (Windows 10.0.0): Executed 3469 of 3470 (skipped 1) SUCCESS (13.108 secs / 5.265 secs)
Safari 8.0.8 (Mac OS X 10.10.5): Executed 3466 of 3467 (skipped 1) SUCCESS (6.929 secs / 3.1 secs)
Edge 15.15063.0 (Windows 10.0.0): Executed 3466 of 3467 (skipped 1) SUCCESS (9.859 secs / 5.045 secs)
Chrome 61.0.3163 (Windows 10.0.0): Executed 3469 of 3470 (skipped 1) SUCCESS (7.957 secs / 3.263 secs)
Firefox 57.0.0 (Windows 10.0.0): Executed 3469 of 3470 (skipped 1) SUCCESS (10.407 secs / 3.897 secs)
Safari 9.1.3 (Mac OS X 10.11.6): Executed 3466 of 3467 (skipped 1) SUCCESS (28.652 secs / 18.286 secs)
TOTAL: 27740 SUCCESS

[15:50:04] Finished 'test' after 10 min
[15:50:04] Finished 'test' after 10 min
Too long with no output (exceeded 10m0s)```

**Comment by jsnellbaker** [comment]
@AlessandroDG It looks like the CircleCI job is running in a different instance than the `Prebid`.  When I click on the details link above, it shows the test was ran under `simplaex` (in the top-left corner).  As such, I can't rebuild the job either due to the permissions.

This sometimes happens when the submitter has their own setup with circleCI,  To my knowledge, we haven't found a good fix for this situation.  

I suppose in this case the reviewer will just have to run the tests locally to verify everything's working fine.

**Comment by AlessandroDG** [comment]
@jsnellbaker thanks for getting back so quickly. I am of course available in case you need me to take any actions.

File: src/AnalyticsAdapter.js [hunk 1]
```diff
@@ -16,7 +16,8 @@ const {
     BID_ADJUSTMENT,
     BIDDER_DONE,
     SET_TARGETING,
-    AD_RENDER_FAILED
+    AD_RENDER_FAILED,
+    ADD_AD_UNITS
```
Comment (jsnellbaker) [inline src/AnalyticsAdapter.js]: Can you clarify on the changes in this file?  

If a new analytics event is needed, that should be part of a separate PR (since it's a core change to the module, rather than just your adapter).

Comment (AlessandroDG) [reply src/AnalyticsAdapter.js]: Sure, 

The event `addAdUnits` helps us detect any change on the adUnits object between the moment a new value is assigned to it and any other moment in the flow.

We are currently handling it at a lower level with:

```    
    pbjsGlobal.onEvent('addAdUnits', (data) => { myHandler(data); });
```

but I understood that the events we get from it are not considered a stable interface.

I will create a separate PR fo it.

Comment (AlessandroDG) [reply src/AnalyticsAdapter.js]: @jsnellbaker I updated this PR. Now it is only about rivrAnalyticsAdapter.

Comment (AlessandroDG) [reply src/AnalyticsAdapter.js]: @jsnellbaker  PR for changes in core have been moved to #3691

**Comment by jsnellbaker** [comment]
@AlessandroDG This PR looks good.  

Do you have a preference though when this merged in lieu of https://github.com/prebid/Prebid.js/pull/3691?  Can this go separate, or do you want the other item merged first before updating your adapter?

**Comment by AlessandroDG** [comment]
@jsnellbaker thanks for asking.
Feel free to merge this PR and #3691 in any order.
